Branch-SDK/src/io/branch/referral/InstallListener.java

```diff
@@ -52,6 +66,97 @@ public void run() {
     @Override
     public void onReceive(Context context, Intent intent) {
         String rawReferrerString = intent.getStringExtra("referrer");
+        processReferrerInfo(context, rawReferrerString, 0, 0);
+        if (isWaitingForReferrer && !isReferrerClientAvailable) { // Still wait for referrer client to get the time stamps if it is active
+            reportInstallReferrer();
+        }
+    }
+    
+    private static void onReferrerClientFinished(Context context, String rawReferrerString, long clickTS, long InstallBeginTS) {
+        processReferrerInfo(context, rawReferrerString, clickTS, InstallBeginTS);
+        if (isWaitingForReferrer) {
+            reportInstallReferrer();
+        }
+    }
+    
+    private static void onReferrerClientError() {
+        isReferrerClientAvailable = false;
```

Comment:
don't we need to handle the callback here in the error case?

Reply:
Referrr client is not support on many play store app versions yet. So dev need to have the  broadcast listener at least for some more time. In case of ReferrerClientError we still need to check if we can get the referrer from broadcast.
Callback happens when the  time out is triggered in the above scenario

Reply:
Understood!